bundles/org.openhab.binding.asuswrt/src/main/java/org/openhab/binding/asuswrt/internal/structures/AsuswrtTraffic.java

```diff
@@ -0,0 +1,202 @@
+        } else {
+            logger.trace("({}) can't set Trafficdata", representationProperty);
+        }
+        this.lastUpdate = System.currentTimeMillis();
+        setZeroHourTraffic(this.totalRX, this.totalTX);
+    }
+
+    /**
+     * Set Zero Hour Traffic.
+     * Save traffic value at new day
+     * 
+     * @param totalRX
+     * @param totalTX
+     */
+    private void setZeroHourTraffic(Integer totalRX, Integer totalTX) {
+        Date now = new Date();
+        if (!AsuswrtUtils.isSameDay(now, this.zeroHourDate) || this.zeroHourRX > totalRX) {
+            this.zeroHourRX = totalRX;
+            this.zeroHourTX = totalTX;
+            this.zeroHourDate = now;
+        }
+    }
+
+    /**
+     * Get value from HEXvalue
+     * 
+     * @param jsonObject jsonObejct has values
+     * @param jsonMember name of jsonMember value is stored
+     * @return
+     */
+    private Integer getTrafficFromHex(JsonObject jsonObject, String jsonMember) {
+        Long lngVal;
+        if (jsonObject.has(jsonMember)) {
+            String hex = jsonObjectToString(jsonObject, jsonMember);
+            try {
+                lngVal = Long.decode(hex);
+                lngVal = lngVal * 8 / 1024 / 1024 / 2;
+                return lngVal.intValue();
+            } catch (Exception e) {
+                logger.debug("({}) error calculating traffic from hex '{}'", representationProperty, hex);
+            }
+        }
+        return -1;
+    }
+
+    /**
+     * Calculate Traffic from Total Traffic and Time
+     * 
+     * @param actVal actual Value to calculate
+     * @param oldVal old Value to calculate
+     * @return
+     */
+    private Double calculateCurrentTraffic(Integer actVal, Integer oldVal) {
+        if (this.lastUpdate > 0) {
+            Long timeSpan = (System.currentTimeMillis() - this.lastUpdate) / 1000;
+            Integer div = 0;
+            try {
+                if (actVal >= 0) {
+                    div = actVal - oldVal;
+                    return Double.valueOf(div / timeSpan.intValue());
+                }
+            } catch (Exception e) {
+                logger.debug("({}) error calculating traffic from timeSpan '{}/{}'", representationProperty, div,
+                        timeSpan);
+            }
+        }
+        return -1.0;
+    }
+
+    /***********************************
+     *
+     * GET VALUES
+     *
+     ************************************/
+
+    public Double getCurrentRX() {
+        return this.curRX;
+    }
+
+    public Double getCurrentTX() {
+        return this.curTX;
+    }
+
+    public Integer getTotalRX() {
+        return this.totalRX;
+    }
+
+    public Integer getTotalTX() {
+        return this.totalTX;
+    }
+
+    public Integer getTodayRX() {
+        return this.totalRX - this.zeroHourRX;
+    }
+
+    public Integer getTodayTX() {
+        return this.totalTX - this.zeroHourTX;
```

Comment:
It would result in less and easier to read code if you remove all the redundant references to `this.` from the code.